builtin/providers/aws/resource_aws_vpc_dhcp_options.go

```diff
@@ -0,0 +1,280 @@
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"netbios_name_servers": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"tags": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsVpcDhcpOptionsCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	setDHCPOption := func(key string) *ec2.NewDHCPConfiguration {
+		log.Printf("[DEBUG] Setting DHCP option %s...", key)
+		tfKey := strings.Replace(key, "-", "_", -1)
+
+		value, ok := d.GetOk(tfKey)
+		if !ok {
+			return nil
+		}
+
+		if v, ok := value.(string); ok {
+			return &ec2.NewDHCPConfiguration{
+				Key: aws.String(key),
+				Values: []*string{
+					aws.String(v),
+				},
+			}
+		}
+
+		if v, ok := value.([]interface{}); ok {
+			var s []*string
+			for _, attr := range v {
+				s = append(s, aws.String(attr.(string)))
+			}
+
+			return &ec2.NewDHCPConfiguration{
+				Key:    aws.String(key),
+				Values: s,
+			}
+		}
+
+		return nil
+	}
+
+	createOpts := &ec2.CreateDHCPOptionsInput{
+		DHCPConfigurations: []*ec2.NewDHCPConfiguration{
+			setDHCPOption("domain-name"),
+			setDHCPOption("domain-name-servers"),
+			setDHCPOption("ntp-servers"),
+			setDHCPOption("netbios-node-type"),
+			setDHCPOption("netbios-name-servers"),
+		},
+	}
+
+	resp, err := conn.CreateDHCPOptions(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DHCP Options Set: %s", err)
+	}
+
+	dos := resp.DHCPOptions
+	d.SetId(*dos.DHCPOptionsID)
+	log.Printf("[INFO] DHCP Options Set ID: %s", d.Id())
+
+	// Wait for the DHCP Options to become available
+	log.Printf("[DEBUG] Waiting for DHCP Options (%s) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"pending"},
+		Target:  "",
+		Refresh: DHCPOptionsStateRefreshFunc(conn, d.Id()),
+		Timeout: 1 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf(
+			"Error waiting for DHCP Options (%s) to become available: %s",
+			d.Id(), err)
+	}
+
+	return resourceAwsVpcDhcpOptionsUpdate(d, meta)
+}
+
+func resourceAwsVpcDhcpOptionsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	req := &ec2.DescribeDHCPOptionsInput{
+		DHCPOptionsIDs: []*string{
+			aws.String(d.Id()),
+		},
+	}
+
+	resp, err := conn.DescribeDHCPOptions(req)
+	if err != nil {
+		return fmt.Errorf("Error retrieving DHCP Options: %s", err)
+	}
+
+	if len(resp.DHCPOptions) == 0 {
+		return nil
+	}
+
+	opts := resp.DHCPOptions[0]
+	d.Set("tags", tagsToMapSDK(opts.Tags))
+
+	for _, cfg := range opts.DHCPConfigurations {
+		tfKey := strings.Replace(*cfg.Key, "-", "_", -1)
+
+		if _, ok := d.Get(tfKey).(string); ok {
+			d.Set(tfKey, cfg.Values[0].Value)
+		} else {
+			values := make([]string, 0, len(cfg.Values))
```

Comment:
I all the testing I've done AWS has been consistently returning them in the same order I defined them. However, I agree with sorting the values just in case.